data_steward/tools/clean_test_project.py

```diff
@@ -0,0 +1,106 @@
+# Python imports
+import argparse
+import logging
+import sys
+
+# Third party imports
+from googleapiclient.errors import HttpError
+
+# Project imports
+from utils import bq
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+
+
+def get_datasets_with_substrings(datasets_list, name_substrings):
+    """
+    Filters list of datasets with specified substrings (e.g. github usernames) in them
+
+    :param datasets_list: list of dataset_ids
+    :param name_substrings: identifies substrings that help identify datasets to delete
+    :return: list of dataset_ids with any substring in their dataset_id
+    """
+    datasets_with_substrings = []
+    for dataset in datasets_list:
+        if any(name_substring in dataset for name_substring in name_substrings):
+            datasets_with_substrings.append(dataset)
+    return datasets_with_substrings
+
+
+def delete_datasets(project_id, datasets_to_delete_list):
+    """
+    Deletes datasets using their dataset_ids
+
+    :param project_id: identifies the project
+    :param datasets_to_delete_list: list of dataset_ids to delete
+    :return:
+    """
+    failed_to_delete = []
+    for dataset in datasets_to_delete_list:
+        try:
+            bq.delete_dataset(project_id, dataset)
+            logging.info(f'Deleted dataset {dataset}')
+        except HttpError:
+            logging.exception(f'Could not delete dataset {dataset}')
+            failed_to_delete.append(dataset)
+    logging.info(
+        f'The following datasets could not be deleted: {failed_to_delete}')
+
+
+def run_deletion(project_id, name_substrings):
+    """
+    Deletes datasets from project containing any of the name_substrings
+
+    :param project_id: identifies the project
+    :param name_substrings: Identifies substrings that help identify datasets to delete
+    :return:
+    """
+    all_datasets = [
+        dataset.dataset_id for dataset in bq.list_datasets(project_id)
+    ]
+    datasets_with_substrings = get_datasets_with_substrings(
+        all_datasets, name_substrings)
+    logging.info(f'Datasets marked for deletion: {datasets_with_substrings}')
+    logging.info('Proceed?')
+    response = get_response()
+    if response == "Y":
+        delete_datasets(project_id, datasets_with_substrings)
+    elif response.lower() == "n":
+        logging.info("Aborting deletion")
+
+
+# Make sure user types Y to proceed
+def get_response():
+    """Return input from user denoting yes/no"""
+    prompt_text = 'Please press Y/n\n'
+    response = input(prompt_text)
+    while response not in ('Y', 'n', 'N'):
+        response = input(prompt_text)
+    return response
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=
+        'Deletes datasets containing specific strings in the dataset_id.',
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('-p',
```

Comment:
One final thing.  This file is called `clean_test_project`.  But it could be applied to any project and its datasets.  Especially because we are allowing the project_id and name substrings to be passed as arguments to the script.  So a more better name might be `clean_project` or `clean_project_datasets`.